Plugins page UX improvement

mariusandra opened the pull request:
## Changes

- Followup on #2939
- Previously if a plugin was installed, but never configured, the switch in the beginning of the line was disabled without an explanation
- Now it will work normally, but instead of directly enabling the plugin it will open the configuration drawer so you could update the required fields, etc
![image](https://user-images.githubusercontent.com/53387/105147265-3d82a880-5b01-11eb-9585-04169d1f3296.png)


## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

mariusandra committed:
open drawer if plugin installed but never configured when toggling switch

mariusandra committed:
?.

Twixes reviewed (commented):
I suggest a small improvement to the UX (not sure how involved state management-wise): when you click "Yes" in "Are you sure you wish to enable this plugin?" and then the drawer opens because configuration is needed, the plugin should be Enabled in the drawer – which is a switch independent from the one in the Installed list, and which will be saved along with configuration. I was a bit confused otherwise.
And one tiny thing: it's "Learn More" in drawer and "Learn more" otherwise, we could standardize.

BTW, review app works! Can we close PostHog/plugin-server#104?

mariusandra committed:
set initial/unsaved changes when opening plugin drawer

mariusandra committed:
set "enabled" to true when installing plugin

mariusandra commented:
Good idea! I added this now. I also added this to the flow when you first install a plugin. The drawer will also have enabled, ahem, enabled. If you then click "cancel" the plugin won't be enabled.

I did that because I even found myself forgetting to toggle it every now and then as well after installing a plugin.

mariusandra committed:
use plugin url as key to remove installed plugins from repository, not name

mariusandra commented:
One more tweak. I'm now removing the plugins from the repository based on their URLs, not names. The names were different in the repository `.json` and after installation in `plugin.json`, so we couldn't use them to match the installed plugins.

mariusandra committed:
extra fix

mariusandra committed:
disallow multiple plugin installations from the same URL

mariusandra commented:
The same block is now updated in the backend as well. Ready for another review.

Twixes reviewed (changes requested):
Looks good! Just one point: "display" names are shown in Available when possible, but only ever "raw" names in Installed.

mariusandra commented:
Well, I would actually just do point 5 [from here](https://github.com/PostHog/plugin-server/issues/13#issuecomment-763452895) and make the "name" field in `plugin.json` mean "display name". 

But I guess that means we're deadlocked here as well? :D

Twixes reviewed (approved):
Let's just go ahead with we what have now then.
